pkg/kubernetes/pkg/apis/componentconfig/types.go

```diff
@@ -43,7 +43,4 @@ type LeaderElectionConfiguration struct {
 	// acquisition and renewal of a leadership. This is only applicable if
 	// leader election is enabled.
 	RetryPeriod metav1.Duration
-	// resourceLock indicates the resource object type that will be used to lock
-	// during leader election cycles.
-	ResourceLock string
```

Comment:
We don't want to make changes to files under pkg/kubernetes. Those files are copied directly from kube/kube. See https://github.com/kubernetes-incubator/service-catalog/blob/master/pkg/kubernetes/README.md.

Reply:
Ok, that makes sense. But i'd like to remove the option anyway (another change - see previous comment on the page). Is that ok? I think having an option that does not work is not great and misleading. That actually made me send this pr - I looked at the code and realized that even thought there is this option, it is ignored. Another approach would be to properly support this option but it feels unnecessary... WDYT?

Reply:
Why not just call https://github.com/kubernetes/client-go/blob/72e1c2a1ef30b3f8da039e92d4a6a1f079f374e8/tools/leaderelection/resourcelock/interface.go#L79 like kube-controller-manager does?

Reply:
The helm chart would set up the rbac for ConfigMaps and set resourceLock to configmaps. If a user wanted to use endpoints instead, they could, but they would need to set up the rbac themselves.